### DISCUSSION TITLE
Probleme with Player Info and ping

### DISCUSSION THREAD

**Discussion by MrBretze** [Minestom/Minestom#851] [Help]
Hello, i have an a issue with the PlayerInfo packet with the action '2', the best ping that i get in solo is ~60ms but in minecraft vanilla is ~1-5ms

i use the last version (446e4a64b7)

**Comment by Eoghanmc22** [comment]
This is related to how minestom processes packets. it will send the keep alive packets at the end of a tick and then the soonest it could process the responce is the next tick. this results in the ping being in units of roughly 50ms. (packets are read async but processed sync) It could be a good idea to process some packets async tho
